### PR TITLE
docs(memory): add README

### PR DESCRIPTION
**Pull request by hekike**
I started to move the [user journey's doc](https://docs.google.com/document/d/1fAvQyelASBK5qkZiF4CP9LXqWxfAAhhSh37iBNKb-GQ/edit#heading=h.c521enci12m1) content to our new documentation structure.

### CODE REVIEW

File: documentation/memory/step2/using_heap_snapshot.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Using Heap Snapshit
```
Comment (cjihrig) [inline documentation/memory/step2/using_heap_snapshot.md]:
```suggestion
# Using Heap Snapshot
```

**Comment by gireeshpunathil** [comment]
landed in c889b62a643ec9a5df3b6e019bc6b4137c334a5b

File: documentation/memory/step2/using_heap_snapshot.md (hunk 1 above)
Comment (gireeshpunathil) [reply documentation/memory/step2/using_heap_snapshot.md]: @cjihrig @hekike - I made this change while landing myself, hope you are good with that!

**Comment by hekike** [comment]
Thank you!

**Pull request closed without merging**